DTXManiaプロジェクト/コード/全体/KeyboardSoundGroupLevelControlHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using FDK;
+
+namespace DTXMania
+{
+    /// <summary>
+    /// KeyboardSoundGroupLevelControlHandler is called by the song selection
+    /// and song play stages when handling keyboard input. By delegating to
+    /// this class they are able to support a centrally-managed and consistent
+    /// set of keyboard shortcuts for dynamically adjusting four sound group
+    /// levels:
+    /// - sound effect level, via Ctrl and either of the Minus or Equals keys
+    /// - voice level, via Shift and either of the Minus or Equals keys
+    /// - song preview and song playback level, via the Minus or Equals key
+    ///
+    /// When the sound group levels are adjusted in this manner, the
+    /// SoundGroupLevelController (and handlers bound to its events) ensure
+    /// that both the sound object group levels are updated and the application
+    /// configuration is updated. See ConfigIniToSoundGroupLevelControllerBinder
+    /// for more details on the latter.
+    /// </summary>
+    internal static class KeyboardSoundGroupLevelControlHandler
```

Comment:
Here is the new KeyboardSoundGroupLevelControlHandler. This is the code, mentioned earlier, that centrally handles sound group level changes caused by keyboard shortcuts during song select and song play. Doc comment and code are well worth a read.

Note that in here it also decides which system sounds to use to give feedback. I've chosen a decent one for voice level changes but if you'd like me to use a different one then just let me know.